pkg/engine/virtualmachinescalesets.go

```diff
@@ -453,6 +453,14 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 		vmssVMProfile.EvictionPolicy = compute.VirtualMachineEvictionPolicyTypes(fmt.Sprintf("[variables('%sScaleSetEvictionPolicy')]", profile.Name))
 	}
 
+	if profile.IsSpotScaleSet() {
+		vmssVMProfile.Priority = compute.VirtualMachinePriorityTypes(fmt.Sprintf("[variables('%sScaleSetPriority')]", profile.Name))
```

Comment:
the first two assignments here are identical to the assignments in the above `if profile.IsLowPriorityScaleSet() {` condition block. Let's expand that conditional block to be `if profile.IsLowPriorityScaleSet() || profile.IsSpotScaleSet() {`, and remove the duplicate assignment statement.

This way this new `if profile.IsSpotScaleSet() {` conditional block will only have the `BillingProfile` assignment, which is the only assignment here unique to spot scale sets.

Reply:
update the code as suggested.